### ISSUE TITLE
yarn doesn't like slashes in dependency branch names

### ISSUE THREAD

**Issue by Tyriar** [yarnpkg/yarn#744]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

The branch cannot be found with the following dependency in my package.json:

```
"xterm": "git+https://github.com/Tyriar/xterm.js.git#vscode-release/1.6",
```

**If the current behavior is a bug, please provide the steps to reproduce.**

I'm trying to build vscode using yarn. I cloned the repo, copied `./scripts/npm.sh` to `./scripts/yarn.sh` and replaced all instances of "npm" in the new file to "yarn" then ran it:

```
./scripts/yarn.sh
error Couldn't find match for "vscode-release/1.6" in "119_fix_cursor_blink,124_add_textarea_back,124_fix_focus_blur,132_fix_constant_focus,154_fix_resize_at_top_ex
ception,158_add_webpack,159_fix_nbsp_resize,209_hide_textarea_when_not_composing,233_fix_cursor_on_top_of_ime,235_fix_scroll_to_top_on_app_mode,242_enable_scroll_on
_soft_reset,add_linkify_to_demo,addon,fix,full_ts_conversion_wip,gh-pages,improve_copyright_line,issue,linkify,master,package-json,patch-1,patch-2,remove_linkify_fr
om_readme,set_package_main,typescript_build,typescript_build_2,vscode-release,windows_demo,0.10,0.11,0.12,0.13,0.14,0.15,0.16,0.17,0.18,0.19,0.20,0.21,0.22,0.23,0.2
4,0.25,0.26,0.27,0.29,0.3,0.30,0.31,0.32,0.33,0.4,0.5,0.6,0.7,0.8,0.9,0.9.1,0.9.2,0.9.3,v0.0.2" for "https://github.com/Tyriar/xterm.js.git".
```

I imagine this will repro with an empty project with a branch containing a slash.

**What is the expected behavior?**

It should work like with `npm`

**Please mention your node.js, yarn and operating system version.**

node 6.7.0
yarn 0.15.1
Ubuntu 16.04

**Comment by Rurik19** [comment]
It also cannot install new-way-typescript-2 typings, whose named like types@/react

**Comment by ramoona** [comment]
Looks like this bug is already fixed https://github.com/yarnpkg/yarn/pull/813

**Comment by kittens** [comment]
Fixed via #813.
